sunpy/map/maputils.py

```diff
@@ -1,7 +1,259 @@
+        dimensions of the map.
+    """
+    return np.meshgrid(*[np.arange(v.value) for v in smap.dimensions]) * u.pix
+
+
+def all_coordinates_from_map(smap):
+    """
+    Returns the coordinates of every pixel in a map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A SunPy map.
+
+    Returns
+    -------
+    `~astropy.coordinates.SkyCoord`
+        An array of sky coordinates in the coordinate system "coordinate_system".
+    """
+    x, y = all_pixel_indices_from_map(smap)
+    return smap.pixel_to_world(x, y)
+
+
+def map_edges(smap):
+    """
+    Returns the pixel locations of the edges of a rectangular input map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A SunPy map.
+
+    Returns
+    -------
+    `~dict`
+        Returns the pixels of edge of the map.
+    """
+    # Calculate all the edge pixels
+    nx, ny = smap.dimensions.x.value, smap.dimensions.y.value
+    bottom = list(product([0.0], np.arange(nx))) * u.pix
+    top = list(product([ny - 1], np.arange(nx))) * u.pix
+    lhs = list(product(np.arange(ny), [0])) * u.pix
+    rhs = list(product(np.arange(ny), [nx - 1])) * u.pix
+    return {"top": top, "bottom": bottom, "lhs": lhs, "rhs": rhs}
+
+
+def contains_full_disk(smap):
+    """
+    Checks if a map contains the full disk of the Sun.
+
+    A map contains the full disk of the Sun if the following two
+    conditions are met: (1) all the pixels at the edge of the map are
+    more than 1 solar radius from the center of the Sun and, (2) the
+    map is not all off disk. If both these conditions are met, the
+    function returns `True`. Otherwise, the function returns `False`.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A map in helioprojective Cartesian coordinates.
+
+    Returns
+    -------
+    `~bool`
+        Returns `False` if any of the edge pixels are less than one solar
+        radius away from the center of the Sun.
+
+    Notes
+    -----
+    * The function assumes that the input map is rectangular.
+    * In the case of coronagraph images the disk itself need not be observed.
+
+    """
+    # Calculate all the edge pixels
+    edges = map_edges(smap)
+    edge_pixels = list(chain.from_iterable([edges["lhs"], edges["rhs"], edges["top"], edges["bottom"]]))
+    x = [p[0] for p in edge_pixels] * u.pix
+    y = [p[1] for p in edge_pixels] * u.pix
+
+    # Calculate the edge of the world
+    edge_of_world = smap.pixel_to_world(x, y)
+
+    # Calculate the distance of the edge of the world in solar radii
+    distance = u.R_sun * np.sqrt(edge_of_world.Tx ** 2 + edge_of_world.Ty ** 2) / smap.rsun_obs
+
+    # Test if all the edge pixels are more than one solar radius distant
+    # and that the whole map is not all off disk.
+    return np.all(distance > 1*u.R_sun) and ~is_all_off_disk(smap)
+
+
+@u.quantity_input
+def coordinate_is_on_disk(coordinate, scale: u.arcsecond):
```

Comment:
This is a strange function. At first it seemed like you wanted to see if a particular HPC near the limb is actually on the Sun which is useful but then scale is also introduced which makes it seem like you actually are talking about pixels? I'm so confused?!

Reply:
Yeah, there is a mistake in the docstring.  The `scale` that is implied in the code is the radius of the Sun in arcseconds.  The docstring should say that.

Reply:
I changed how `coordinate_is_on_disk` works to eliminate the use of scale keyword.